crates/worker/src/partition/state_machine/mod.rs

```diff
@@ -199,10 +182,7 @@ impl<'a, S> StateMachineApplyContext<'a, S> {
             Span::record_invocation_target(invocation_target);
         }
         if let Some(journal_metadata) = status.get_journal_metadata() {
-            journal_metadata
-                .span_context
-                .as_parent()
-                .attach_to_span(&Span::current());
+            Span::current().set_relation(journal_metadata.span_context.as_parent());
```

Comment:
I am a bit unsure about linking internal spans to the services trace. The problem I see is that we will end up with orphaned spans if the runtime and services tracing spans end up on different jaeger endpoints.

Reply:
This is handled during Export of spans (by the RuntimeServicesExporter) to separate (and alter) services spans. 

The exporter will make sure to emit all spans to the 2 endpoints, it will also derive new traces and span ids for services spans so they maintain their relationship if they are reported to the same or different endpoint